src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -1036,7 +1034,10 @@ HeapWord* ShenandoahHeap::allocate_memory(ShenandoahAllocRequest& req) {
         pacer()->unpace_for_alloc(pacer_epoch, requested - actual);
       }
     } else {
-      increase_used(actual*HeapWordSize);
+      if (req.is_old()) {
+        old_generation()->increase_allocated(actual_bytes);
```

Comment:
I'm not sure I follow all the logic.  When we allocate a PLAB for promotions into old-gen, we want to increment the promotion rate.  But when we allocate a ?LAB to hold the results of an old-gen allocation, that should not be counted as "promotion" memory.  Are we distinguishing these two cases?

Reply:
Ah, that's a good catch. I think this would count evacuations in old as part of the promotion rate. Will work on a fix.